Source/Core/VideoBackends/OGL/VertexManager.cpp

```diff
@@ -140,6 +140,15 @@ void VertexManager::vFlush(bool useDstAlpha)
 
 	PrepareDrawBuffers(stride);
 
+	if (!bpmem.genMode.zfreeze && IndexGenerator::GetIndexLen() >= 3)
+	{
+		CalculateZSlope(stride);
```

Comment:
instead of calculating z slope for every draw call that do not use  zfreeze why not calculate zslope on the first draw call that uses zfreeze, and just store the last 3 vertices and mtx data in the other cases?